src/elements/Pagination/index.jsx

```diff
@@ -0,0 +1,205 @@
+          res.push(totalPages)
+        } else if (i === rightLength - 2) {
+          res.push('ellipsis')
+        } else {
+          res.push(currentPage + i + 1)
+        }
+      }
+    } else {
+      for (let i = 0; i < rightLength; i += 1) {
+        res.push(currentPage + i + 1)
+      }
+    }
+    return res;
+  }
+
+  renderPagination () {
+    const { currentPage, totalPages } = this.state
+    if (currentPage < 1) {
+      this.changePage(1)
+    } else if (currentPage > totalPages) {
+      this.changePage(totalPages)
+    }
+
+    const paginations = []
+    const steps = this.calculateSteps();
+
+    for (let i = 0; i < steps.length; i += 1) {
+      const label = steps[i];
+      let isClickable = true;
+
+      if (currentPage === steps[i]) {
+        isClickable = false;
+      }
+      if (label === 'ellipsis') {
+        paginations.push(<PaginationLink isPageBreak={true} key={i} />)
+      } else {
+        paginations.push(
+          <PaginationLink
+            label={label}
+            isClickable={isClickable}
+            onClick={(page) => { this.onClick(page) }}
+            key={i} />
+        )
+      }
+    }
+
+    return paginations;
+  }
+
+  renderLeftCaret () {
+    return (
+      <li className="Pagination__caret">
+        { this.state.currentPage <= 1
+          ? <span>{'<'}</span>
+          : <a onClick={() => {
+            this.onClick(this.state.currentPage - 1)
+          }}>
+            {'<'}
+          </a> }
+      </li>
+    )
+  }
+
+  renderRightCaret () {
+    return (
+      <li className="Pagination__caret">
+        { this.state.currentPage >= this.state.totalPages
+          ? <span>{'>'}</span>
+          : <a onClick={() => {
+            this.onClick(this.state.currentPage + 1)
+          }}>
+            {'>'}
+          </a>
+        }
+      </li>
+    )
+  }
+
+  render () {
+    const leftCaret = this.renderLeftCaret();
+    const rightCaret = this.renderRightCaret();
+    const pagination = this.renderPagination();
+
+    return (
+      <div className={classNames(
+        'Pagination',
+        `Pagination--${this.props.fontSize}`)}>
+        <ul className={classNames('Pagination__list')}>
+          {leftCaret}
+          {pagination}
+          {rightCaret}
+        </ul>
+      </div>
+    )
+  }
+}
+
+Pagination.propTypes = {
+  /** Current selected pagination number */
+  currentPage: PropTypes.number,
+  /** Specify how manys items showed in one page */
+  itemsPerPage: PropTypes.number.isRequired,
+  /** Specify the total items */
+  totalItems: PropTypes.number.isRequired,
+  /** Number of elements in pagination component */
+  maxPaginationSize: numberInRange(5, 20),
+  /** Action to be performed on page change */
+  onChange: PropTypes.func.isRequired,
+  /** Set font size to the pagination */
+  fontSize: PropTypes.string,
+}
+
+Pagination.defaultProps = {
+  currentPage: 1,
+  maxPaginationSize: 7,
+  fontSize: 'medium',
+};
```

Comment:
when you do with class component, you don't need to defined this out of class block. you can do this with
```jsx
class Pagination extends Component {
  static propTypes = {
    /** Current selected pagination number */
    currentPage: PropTypes.number,
    /** Specify how manys items showed in one page */
    itemsPerPage: PropTypes.number.isRequired,
    /** Specify the total items */
    totalItems: PropTypes.number.isRequired,
    /** Number of elements in pagination component */
    maxPaginationSize: numberInRange(5, 20),
    /** Action to be performed on page change */
    onChange: PropTypes.func.isRequired,
    /** Set font size to the pagination */
    fontSize: PropTypes.string,
  }

  static defaultProps = {
    currentPage: 1,
    maxPaginationSize: 7,
    fontSize: 'medium',
  }

  state = {
    this.state = {
      currentPage,
      totalPages: Math.ceil(this.props.totalItems / this.props.itemsPerPage),
      maxPaginationSize,
    }
  }

  onClick (page) {
    let currentPage = page;
 ...
```